ChuckNorrisFacts/Presentation/List Facts/ListFactsViewModel.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  ListFactsViewModel.swift
+//  ChuckNorrisFacts
+//
+//  Created by Munir Wanis on 21/05/18.
+//  Copyright © 2018 Wanis. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+
+class ListFactsViewModel {
+    let service = FactService()
+    
+    var presentation: Observable<[FactPresentation]>
+    
+    init() {
+        presentation = service.getFacts(term: "")
+            .map { facts in ListFactsViewModel.convert(facts) }
+            .observeOn(MainScheduler.instance)        
+    }
+    
+    private static func convert(_ facts: Facts) -> [FactPresentation] {
```

Comment:
Eu tenho uma tendência a colocar esse métodos numa classe separada, porque eles não me parecem uma responsabilidade direta do viewmodel, mas mais um "método auxiliar"... 
E aí não colocaria como método estático também. @ubiratansoares lembro de uma vez você ter falado sobre as desvantagens de ficar usando tudo como método estático (nesse caso, método da classe ao invés de ser da instância), mas não sei dizer a diferença
Eu to lembrando certo mesmo? Consegue resumir pra gente?

Reply:
Eu tinha botado estático para fazer um teste direto no init, já tirei o `static` dele, mas ainda está na mesma classe. Vou talvez o ideal seja criar um `mapper`. Por ser específico para essa tela o `mapper` nesse caso ficaria aonde? Na camada de `Domain` ou `Presentation` mesmo?

Reply:
Vc tá mapeando um objeto do domínio pra um do presentation.... Vulgo, "não sei" 😅
Aí acho que vai mais de gosto seu mesmo

Reply:
Não faz sentido ter um método estático e privado dentro de uma classe. Isso é um anti-pattern que denota que esse método provavelmente pertence a uma outra classe com responsabilidade melhor definida